File: JavaScript/6js 2/towerBuilder.js
/*
|--------------------------------------------------------------------------
| Build Tower
|--------------------------------------------------------------------------
|
*/

function towerBuilder(nFloors) {
   let res = [];
   for (let i = 0; i < nFloors; i++) {
      let x = "*".repeat(2 * i + 1);
      let y = " ".repeat(nFloors - i - 1);
      res.push(y + x + y);
   }
   return res;
}


function towerBuilder(n) {
   return [...Array(nFloors)].map((_, idx) =>
      `${`*`.repeat(2 * idx + 1)}`.padStart(nFloors + idx, ` `).padEnd(2 * nFloors - 1, ` `)
   );
}


function towerBuilder(n) {
   return [...Array(n)].map(
      (_, i) => " ".repeat(n - 1 - i) + "*".repeat(i * 2 + 1) + " ".repeat(n - 1 - i)
   );
}


console.log(towerBuilder(1), ["*"]);
console.log(towerBuilder(2), [" * ", "***"]);
console.log(towerBuilder(3), ["  *  ", " *** ", "*****"]);

File: Python/6py 2/tower_builder.py
"""
|--------------------------------------------------------------------------
| Build Tower
|--------------------------------------------------------------------------
|
"""

def tower_builder(n_floors):
   res = []
   for i in range(n_floors):
      x, y = "*" * (2 * i + 1), " " * (n_floors - i - 1)
      res.append(y + x + y)
   return res


def tower_builder(n):
   return [f'{"*"*i:^{2*n-1}}' for i in range(1,2*n,2)]


def tower_builder(n):
   return [("*" * (i*2-1)).center(n*2-1) for i in range(1, n+1)]


print(tower_builder(1), ['*', ])
print(tower_builder(2), [' * ', '***'])
print(tower_builder(3), ['  *  ', ' *** ', '*****'])